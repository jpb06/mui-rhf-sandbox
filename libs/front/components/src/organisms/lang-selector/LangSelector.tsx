import { Link as MuiLink } from '@mui/material';
import { useTranslation } from 'next-i18next';
import Link from 'next/link';

export const LangSelector = () => {
  const { t } = useTranslation('common');

  return (
    <>
      <Link href="/" locale="en" passHref>
        <MuiLink
          variant="body2"
          sx={{
            marginRight: 1,
            textDecoration: 'none',
          }}
        >
          ğŸ‡¬ğŸ‡§ {t('language.en')}
        </MuiLink>
      </Link>
      |
      <Link href="/" locale="fr" passHref>
        <MuiLink variant="body2" sx={{ marginLeft: 1, textDecoration: 'none' }}>
          ğŸ‡«ğŸ‡· {t('language.fr')}
        </MuiLink>
      </Link>
    </>
  );
};
