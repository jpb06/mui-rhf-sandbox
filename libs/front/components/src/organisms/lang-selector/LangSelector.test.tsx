import { screen, waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

import { render } from '@tests';
import { mockNextRouter, mockUseTranslation } from '@tests/mocks';

import { LangSelector } from './LangSelector';

describe('LangSelector component', () => {
  const { pushMock } = mockNextRouter();

  beforeAll(() => {
    mockUseTranslation('en');
  });

  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('should match snapshot', () => {
    const { baseElement } = render(<LangSelector />);

    expect(baseElement).toMatchSnapshot();
  });

  it('should redirect the user when language is changed', async () => {
    render(<LangSelector />);

    const button = screen.getByText(/common:language.fr/i);
    userEvent.click(button);

    await waitFor(() => {
      expect(pushMock).toHaveBeenCalledTimes(1);
    });
  });

  it('should do nothing when we click on the current language', async () => {
    render(<LangSelector />);

    const button = screen.getByText(/common:language.en/i);
    userEvent.click(button);

    expect(pushMock).toHaveBeenCalledTimes(0);
  });
});
