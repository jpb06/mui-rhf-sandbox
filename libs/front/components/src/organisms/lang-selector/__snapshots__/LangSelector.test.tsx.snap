// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LangSelector component should match snapshot 1`] = `
.emotion-0 {
  margin: 0;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  font-size: 0.875rem;
  line-height: 1.43;
  letter-spacing: 0.01071em;
  color: #006064;
  -webkit-text-decoration: underline;
  text-decoration: underline;
  text-decoration-color: rgba(0, 96, 100, 0.4);
  margin-right: 8px;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-0:hover {
  text-decoration-color: inherit;
}

.emotion-1 {
  margin: 0;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  font-size: 0.875rem;
  line-height: 1.43;
  letter-spacing: 0.01071em;
  color: #006064;
  -webkit-text-decoration: underline;
  text-decoration: underline;
  text-decoration-color: rgba(0, 96, 100, 0.4);
  margin-left: 8px;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-1:hover {
  text-decoration-color: inherit;
}

<body>
  <div>
    <a
      class="MuiTypography-root MuiTypography-body2 MuiLink-root MuiLink-underlineAlways emotion-0"
      href="/"
    >
      ðŸ‡¬ðŸ‡§ 
      common:language.en
    </a>
    |
    <a
      class="MuiTypography-root MuiTypography-body2 MuiLink-root MuiLink-underlineAlways emotion-1"
      href="/"
    >
      ðŸ‡«ðŸ‡· 
      common:language.fr
    </a>
  </div>
</body>
`;
